src/lib/converter/plugins/GitHubPlugin.ts

```diff
@@ -82,9 +90,13 @@ class Repository {
             });
         }
 
-        out = <ShellJS.ExecOutputReturnValue> ShellJS.exec('git rev-parse --short HEAD', {silent: true});
-        if (out.code === 0) {
-            this.branch = out.stdout.replace('\n', '');
+        if (!this.revision) {
+            out = <ShellJS.ExecOutputReturnValue> ShellJS.exec('git rev-parse --short HEAD', {silent: true});
+            if (out.code === 0) {
+                this.branch = out.stdout.replace('\n', '');
+            }
+        } else if (this.revision) {
+            this.branch = this.revision;
```

Comment:
Yes, we'll definitely make it consistent 😄  I'm not convinced on `branch` either, but it is what was previously used I suppose - even though it was pointing to a ref. Which reminds me, does `--git-ref` make more sense then? And then `reference` in the code?